mender-image-prepare.sh

```diff
@@ -0,0 +1,535 @@
+#!/bin/bash
+
+show_help() {
+cat << EOF
+Usage: $0 [prepare_image | make_sdimg | make_sdcard] [-i|--image | -d|--device -r|--rootsize]
+
+        device - can be /dev/sd* or /dev/mmcblk*
+
+        prepare_image - shrinks existing embedded image
+                        obligatory parameters: --image
+
+        make_sdcard   - formats, partitions and sets the sd card
+                        obligatory parameters: --image & --rootsize
+                        
+                        note: rootsize can found as an output from
+                        'prepare_image' command
+
+        make_sdimg    - composes image file compliant with Mender partition
+                        table scheme
+
+EOF
+}
+
+if [ $# -eq 0 ]; then
+  show_help
+  exit 1
+fi
+
+if ! id | grep -q root; then
+        echo "must be run as root"
```

Comment:
I am not really comfortable to require to run the whole script as root.  Things like this become very dangerous:

    rm -rf $BASE_DIR/mnt

Maybe just `sudo` the commands that actually require elevated privileges

Reply:
+1